VPLAY-9947: Fix compilation issues in MAC Simulator

Vinish100 opened the pull request:
Reason for change: Enforce C++ version to be used in the CMakeLists.txt files throughout AAMP to avoid compilation
errors. Bump up CMake required version to 3.5 which is required in latest MAC OS.
Add env variable for building AAMP dependencies in MAC without patching source files.
Refactor CMakeLists.txt. Add option to downgrade asio dependent library to 1.18.2 in MAC for
building subtec. Fix device build errors and L1.
Test Procedure: Compilation should be successful in all platforms
Risks: Low

Vinish100 committed:
VPLAY-9947: Fix compilation issues in MAC Simulator

Reason for change: Enforce C++ version to be used in the
CMakeLists.txt files throughout AAMP to avoid compilation
errors. Bump up CMake required version to 3.5 which is required
in latest MAC OS. Add env variable for building AAMP dependencies
in MAC without patching source files
Test Procedure: Compilation should be successful in all platforms
Risks: Low

Signed-off-by: Vinish100 <vinish.balan@gmail.com>

Vinish100 committed:
VPLAY-9947: Fix compilation issues in MAC Simulator

Reason for change: Refactor CMakeLists.txt. Add option
to downgrade asio dependent library to 1.18.2 in MAC for
building subtec. Fix device build errors and L1.
Test Procedure: Compilation should be successful in all platforms
Risks: Low

Signed-off-by: Vinish100 <vinish.balan@gmail.com>

rdkcmf-jenkins commented:
b'## **Blackduck scan failure details**

**Summary**: 2 **violation**s, 0 files pending approval, 0 files pending identification.
* Protex Server Path: /home/blackduck/github/aamp/225/rdk/components/generic/aamp

* Commit: 70cc5f94b83a488c8bb9d8fd3b536b93b9add032

 **Report detail:** [gist](https://gist.github.com/rdkcmf-jenkins/1730b9b2e6cb5bd466b0206bc48d6457)'

rdkcmf-jenkins commented:
b'## **WARNING:** A Blackduck scan failure has been waived

A prior failure has been upvoted
* Upvote reason: Cleared

* Commit: 70cc5f94b83a488c8bb9d8fd3b536b93b9add032
'

Vinish100 committed:
VPLAY-9947: Fix compilation issues in MAC Simulator

Reason for change: Fix identation and add license headers.
Fix L2 test failure in Ubuntu.
Test Procedure: Compilation should be successful in all platforms
Risks: Low

Signed-off-by: Vinish100 <vinish.balan@gmail.com>

rdkcmf-jenkins commented:
b'## **Blackduck scan failure details**

**Summary**: 2 **violation**s, 0 files pending approval, 0 files pending identification.
* Protex Server Path: /home/blackduck/github/aamp/225/rdk/components/generic/aamp

* Commit: 403cb6c35f0f6cf6d5fe5d2d065ca690951e1cbb

 **Report detail:** [gist](https://gist.github.com/rdkcmf-jenkins/62a0df25137f34df0b788bfd346d75bd)'

jfagunde commented on CMakeLists.txt:
```diff
@@ -426,401 +403,309 @@ if(CMAKE_IARM_MGR)
+	endif()
+
+	if(CMAKE_USE_OPENCDM_ADAPTER_MOCKS)
+		# copy_staging_header_file(open_cdm.h)
+		# copy_staging_header_file(opencdmMocks.cpp)
```
Can these be removed? It seems like commented out old code.

Vinish100 replied:
Done

jfagunde commented on CMakeLists.txt:
```diff
@@ -426,401 +403,309 @@ if(CMAKE_IARM_MGR)
-	set_target_properties(aampKotlin PROPERTIES COMPILE_FLAGS "${LIBAAMP_DEFINES} ${AAMP_CLI_EXTRA_DEFINES} ${OS_CXX_FLAGS}")
-	xcode_define_schema(aampKotlin)
+if (COVERAGE_ENABLED)
+	set(LIBAAMP_DEFINES "${LIBAAMP_DEFINES} --coverage")
+	set(LIBAAMP_DEPENDS ${LIBAAMP_DEPENDS} "--coverage")
```
I'd try to be consistent with the quotes ["] Unless there is a reason to change it, I suggest not to change it as part of this big PR. I.e. leave the code as it is:
`set(LIBAAMP_DEPENDS "${LIBAAMP_DEPENDS} --coverage")`

Vinish100 replied:
I got this suggestion from Copilot actually
The difference is in how CMake handles the variable expansion and whitespace:

1. Quoted Version
Everything inside the quotes is treated as a single string.
If LIBAAMP_DEPENDS is "foo bar" and CMAKE_THREAD_LIBS_INIT is -lpthread,
the result is one list item:
When used in target_link_libraries, this is treated as a single library name (which will likely fail).
2. Unquoted Version
Each variable is expanded and split into list elements.
If LIBAAMP_DEPENDS is "foo;bar" and CMAKE_THREAD_LIBS_INIT is -lpthread,
the result is a list of three items:
This is the correct way to build up a list of libraries for linking.

For libraries it looks like Unquoted is better as each libraries are treated separately. Do let me know your thoughts on this? For DEFINES, this doesn't make much difference, hence went with quoted version

jfagunde commented on CMakeLists.txt:
```diff
@@ -426,401 +403,309 @@ if(CMAKE_IARM_MGR)
-    set(LIBAAMPJSBINDINGS_DEFINES "${LIBAAMPJSBINDINGS_DEFINES} -DUSE_SYSTEMD_JOURNAL_PRINT=1 -DSD_JOURNAL_SUPPRESS_LOCATION=1")
-
+	message("CMAKE_SYSTEMD_JOURNAL set")
+	set(LIBAAMP_DEPENDS ${LIBAAMP_DEPENDS} "-lsystemd")
```
Same comment as above, here and in other lines below.

jfagunde commented on install-aamp.sh:
```diff
@@ -23,6 +23,10 @@ if [[ -z "${MAKEFLAGS}" ]]; then
     export MAKEFLAGS=-j$(nproc)
 fi
 
+if [[ "$OSTYPE" == "darwin"* ]]; then
```
Does this have to be specific for OSX? If possible, would it make sense to export it for ubuntu as well?

Vinish100 committed:
VPLAY-9947: Fix compilation issues in MAC Simulator

Reason for change: Skip building glib-2.0 manually and fixed
build issues in Ubuntu 24.04, address review comments.
Test Procedure: Compilation should be successful in all platforms
Risks: Low

Signed-off-by: Vinish100 <vinish.balan@gmail.com>

jfagunde reviewed (commented):
Looks good to me :)

rdkcmf-jenkins commented:
b'## **Blackduck scan failure details**

**Summary**: 2 **violation**s, 0 files pending approval, 0 files pending identification.
* Protex Server Path: /home/blackduck/github/aamp/225/rdk/components/generic/aamp

* Commit: 23f9f5963aed706e01865f238c9da3c4b16aa6e1

 **Report detail:** [gist](https://gist.github.com/rdkcmf-jenkins/7ab62799eb92b06757b86e6ff2bd5aa7)'

pstroffolino committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-9947_MacBuildIssue

rdkcmf-jenkins commented:
b'## **Blackduck scan failure details**

**Summary**: 2 **violation**s, 0 files pending approval, 0 files pending identification.
* Protex Server Path: /home/blackduck/github/aamp/225/rdk/components/generic/aamp

* Commit: 1b3359fa135a1c7f5afaf096e7ee8d502f7512c0

 **Report detail:** [gist](https://gist.github.com/rdkcmf-jenkins/2c2d488dca9eda0459e78d747b190118)'

rdkcmf-jenkins commented:
b'## **Blackduck scan failure details**

**Summary**: 2 **violation**s, 0 files pending approval, 0 files pending identification.
* Protex Server Path: /home/blackduck/github/aamp/225/rdk/components/generic/aamp

* Commit: 1b3359fa135a1c7f5afaf096e7ee8d502f7512c0

 **Report detail:** [gist](https://gist.github.com/rdkcmf-jenkins/be9810fb92fff04cef069f8dabddb646)'
